### ISSUE TITLE
CI: Warning `The kernel does not support --service-no-backend-response=reject`

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#29733]
We have this warning in many tests of the ginkgo CI:
```
The kernel does not support --service-no-backend-response=reject, falling back to --service-no-backend-response=drop
```

From Slack discussion with Dylan:

> > What's the concrete impact of a disablement? Does it really warrant a warning?
>
> I would have to look it up, but if I recall correctly, we were not conformant to some k8s network test because we were dropping and we are supposed to return a nice ICMP. So when doing the fallback, we technically slip back into non conformance. (I don't know if we care enough to alert on this)
Otherwise, I think it comes back to, are we doing what users ask of us. The reason for this feature was that some users care very much about this behavior to the point that it can be critical. So when we have to fallback, a warning is nice. However, the way the setting is now, we can't detect if its the default value or explicitly set. So what we can do is to add a bestEffort value, make it default, and only warn if the setting as explicitly set to reject.

TL;DR. It really warrants a warning and we should fix the configs in the CI instead of removing the warning.

Warning currently allowlisted at https://github.com/cilium/cilium/blob/5a82225cd713ed4480324a3ebfce449196cbbd3c/test/helpers/cons.go#L248C1-L248C41.

**Comment by julianwiedmann** [comment]
drive-by - if [this](https://github.com/cilium/cilium/pull/28157#discussion_r1818545597) holds true, then this warning shouldn't occur anymore on Cilium v1.16+. We only test with kernels 5.2+, or RHEL8 (which also has the needed kernel support).

**Comment by dylandreimerink** [comment]
> drive-by - if https://github.com/cilium/cilium/pull/28157#discussion_r1818545597 holds true

I think it does. That helm comment seems to be wrong, and on 5.2+ we should not get the message.

**Comment by julianwiedmann** [comment]
> > drive-by - if [#28157 (comment)](https://github.com/cilium/cilium/pull/28157#discussion_r1818545597) holds true
> 
> I think it does. That helm comment seems to be wrong, and on 5.2+ we should not get the message.

Let's find out: https://github.com/cilium/cilium/pull/35830.
